### PR TITLE
Fixed #28725 -- Prevented serializing inherited ManyToManyFields in child model.

### PR DESCRIPTION
**Pull request by nadege**

### CODE REVIEW

File: docs/releases/2.2.3.txt [hunk 1]
```diff
@@ -15,3 +15,7 @@ Bugfixes
 
 * Fixed a regression in Django 2.2.2 where auto-reloader crashes with
   ``AttributeError``, e.g. when using ``ipdb`` (:ticket:`30588`).
+
+* Fixed a bug where in multi-table inheritance, serializers output
+  :class:`~django.db.models.ManyToManyField` in both the parent and the child
+  class. (:ticket:`28725`).
```
Comment (felixxm) [inline docs/releases/2.2.3.txt]: This doesn't qualify for a backport, so you can remove release notes.

Comment (nadege) [reply docs/releases/2.2.3.txt]: Ok, should I put that in docs/release/3.0.txt instead ?

Comment (felixxm) [reply docs/releases/2.2.3.txt]: We don't need any release notes. 3.0 is a feature release and we don't mention bugfixes in such releases.

Comment (nadege) [reply docs/releases/2.2.3.txt]: oh ok, thank you :)

File: tests/serializers/tests.py [hunk 2]
```diff
@@ -340,6 +341,13 @@ def test_serialize_proxy_model(self):
         self.assertEqual(base_data, proxy_data.replace('proxy', ''))
         self.assertEqual(base_data, proxy_proxy_data.replace('proxy', ''))
 
+    def test_serialize_inheritance(self):
```
Comment (felixxm) [inline tests/serializers/tests.py]: `test_serialize_inheritance` -> `test_serialize_inherited_m2m`

File: tests/serializers/tests.py [hunk 3]
```diff
@@ -340,6 +341,13 @@ def test_serialize_proxy_model(self):
...
+        child_object = Child.objects.create(parent_data=2, child_data=3)
+        child_object.parent_m2m.add(Parent.objects.create(parent_data=4))
+        child_data = serializers.serialize(self.serializer_name, [child_object])
+        self.assertNotIn('parent_data', child_data)
```
Comment (felixxm) [inline tests/serializers/tests.py]: This assertion is not related with this fix so I would move it to a separate commit.

File: tests/serializers/tests.py [hunk 4]
```diff
@@ -340,6 +341,13 @@ def test_serialize_proxy_model(self):
         self.assertEqual(base_data, proxy_data.replace('proxy', ''))
         self.assertEqual(base_data, proxy_proxy_data.replace('proxy', ''))
 
+    def test_serialize_inheritance(self):
+        child_object = Child.objects.create(parent_data=2, child_data=3)
+        child_object.parent_m2m.add(Parent.objects.create(parent_data=4))
```
Comment (felixxm) [inline tests/serializers/tests.py]: `parent_data` and `child_data` are `CharField`\`s, so maybe `'a'`, `'b'`, ... etc.

File: tests/serializers/tests.py [hunk 5]
```diff
@@ -340,6 +341,13 @@ def test_serialize_proxy_model(self):
...
+        child_data = serializers.serialize(self.serializer_name, [child_object])
+        self.assertNotIn('parent_data', child_data)
+        self.assertNotIn('parent_m2m', child_data)
```
Comment (felixxm) [inline tests/serializers/tests.py]: `assertNotIn` may pass from many reasons. I think it is better to check field value with `self._get_field_values()` hook, e.g.
```python
self.assertEqual(self._get_field_values(child_data, 'parent_m2m'), [])
```
